VMBackup/main/freezesnapshotter.py

```diff
@@ -251,6 +253,116 @@ def freeze(self):
+                self.logger.log("FS : GetBlobMetadata: HttpCallGetResponse : result :" + str(result) + ", errMsg :" + str(errMsg))
+                blobProperties = self.httpresponse_get_blob_properties(httpResp)
+            except Exception as e:
+                self.logger.log("FS : GetBlobMetadata: Failed to get blob properties with error: %s, stack trace: %s" % (str(e), traceback.format_exc()))
+        return blobProperties
+
+    def GetHeaderSize(self, headers):
+    # max size of blob metadata 
+            return sys.getsizeof(json.dumps(headers))
+
+
+    def httpresponse_get_blob_properties(self, httpResp):
+        blobProperties = {}
+        if(httpResp != None):
+            self.hutil.log("httpresponse_get_blob_properties: Blob-properties response status:"+str(httpResp.status))
+            if(httpResp.status == 200):
+                resp_headers = httpResp.getheaders()
+                blobProperties = resp_headers
+        return blobProperties
+
+    def populate_snapshotreq_headers_perblob(self, sasuri, sasuri_index, backup_meta_data, blobMetadataTelemetryMessage):
+        headers= {}
+        headers["Content-Length"] = '0'
+        blobMetdataMaxSizeBytes = CommonVariables.blobMetdataMaxSizeBytes
+
+        original_blob_metadata = self.GetBlobProperties(sasuri)
+        
+        if(original_blob_metadata is not None): 
+            for meta in original_blob_metadata:
+                Key,Value = meta
+                headers[Key] = Value
+
+        if(backup_meta_data is not None):
+            for meta in backup_meta_data:
+                key = meta['Key']
+                value = meta['Value']
+                headers["x-ms-meta-" + key] = value
+        
+        level1BlobMetadataSize = self.GetHeaderSize(headers)
+        
+        if level1BlobMetadataSize > blobMetdataMaxSizeBytes:
+            if sasuri_index not in blobMetadataTelemetryMessage :
+                blobMetadataTelemetryMessage[sasuri_index] = ""
+
+            blobMetadataTelemetryMessage[sasuri_index]+= str(level1BlobMetadataSize) + ", ";
+
+            headers = {}
+            if(original_blob_metadata is not None): 
+                for meta in original_blob_metadata:
+                    Key,Value = meta
+                    if Key == "x-ms-meta-diskencryptionsettings" :
+                        headers[Key] = Value
+                        break
+
+            if(backup_meta_data is not None):
+                for meta in backup_meta_data:
+                    key = meta['Key']
+                    value = meta['Value']
+                    headers["x-ms-meta-" + key] = value
+
+            level2BlobMetadataSize = self.GetHeaderSize(headers)
+            
+            if level2BlobMetadataSize > blobMetdataMaxSizeBytes :
+                blobMetadataTelemetryMessage[sasuri_index]+= str(level2BlobMetadataSize) + ", ";
+
+                headers = {}
+                if(backup_meta_data is not None):
```

Comment:
suggestion: as discussed offline, can you refactor it by keeping CRP headers as default and make decisions by getting sizes individually instead of always creating final metadata size